nixos/modules/services/networking/opkssh/opkssh.nix

```diff
@@ -0,0 +1,171 @@
+            };
+          };
+        }
+      );
+      default = {
+        google = {
+          issuer = "https://accounts.google.com";
+          clientId = "206584157355-7cbe4s640tvm7naoludob4ut1emii7sf.apps.googleusercontent.com";
+          lifetime = "24h";
+        };
+        microsoft = {
+          issuer = "https://login.microsoftonline.com/9188040d-6c67-4c5b-b112-36a304b66dad/v2.0";
+          clientId = "096ce0a3-5e72-4da8-9c86-12924b294a01";
+          lifetime = "24h";
+        };
+      };
+      description = "OpenID Connect providers configuration";
+      example = literalExpression ''
+        {
+          google = {
+            issuer = "https://accounts.google.com";
+            clientId = "206584157355-7cbe4s640tvm7naoludob4ut1emii7sf.apps.googleusercontent.com";
+            lifetime = "24h";
+          };
+          github = {
+            issuer = "https://token.actions.githubusercontent.com";
+            clientId = "github";
+            lifetime = "oidc";
+          };
+        }
+      '';
+    };
+
+    authorizations = mkOption {
+      type = types.listOf (
+        types.submodule {
+          options = {
+            user = mkOption {
+              type = types.str;
+              description = "Linux user to authorize";
+            };
+
+            principal = mkOption {
+              type = types.str;
+              description = "Principal identifier (email, repo, etc.)";
+            };
+
+            issuer = mkOption {
+              type = types.str;
+              description = "Issuer URI";
+            };
+          };
+        }
+      );
+      default = [ ];
+      description = "User authorization mappings";
+      example = literalExpression ''
+        [
+          {
+            user = "github";
+            principal = "example@gmail.com";
+            issuer = "https://accounts.google.com";
+          }
+          {
+            user = "github";
+            principal = "repo:NixOs/nixpkgs:environment:production";
+            issuer = "https://token.actions.githubusercontent.com";
+          }
+        ]
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.groups.${cfg.group} = { };
+    users.users.${cfg.user} = {
+      isSystemUser = true;
+      description = "OpenPubkey OpenID Connect SSH User";
+      group = cfg.group;
+    };
+
+    services.openssh = {
+      authorizedKeysCommand = "/run/wrappers/bin/opkssh verify %u %k %t";
+      authorizedKeysCommandUser = cfg.user;
+    };
+
+    security.wrappers."opkssh" = {
+      source = "${cfg.package}/bin/opkssh";
+      owner = "root";
+      group = "root";
+    };
+
+    environment.etc."opk/providers" = {
+      mode = "0640";
+      user = cfg.user;
+      group = cfg.group;
+      source = providerFile;
+    };
+
+    environment.etc."opk/auth_id" = {
+      mode = "0640";
+      user = cfg.user;
+      group = cfg.group;
+      source = authIdFile;
+    };
+  };
+}
```

Comment:
ah, forgot to also mention this module needs `maintainers` added to it

Reply:
@johnrichardrinehart @sarcasticadmin would you liked to be added as maintainers for this service as well?